packages/web/src/components/track/GiantTrackTile.tsx

```diff
@@ -0,0 +1,655 @@
+import Badge from './Badge'
+import { CardTitle } from './CardTitle'
+import GiantArtwork from './GiantArtwork'
+import styles from './GiantTrackTile.module.css'
+import { GiantTrackTileProgressInfo } from './GiantTrackTileProgressInfo'
+import InfoLabel from './InfoLabel'
+import { PlayPauseButton } from './PlayPauseButton'
+import { PremiumTrackSection } from './PremiumTrackSection'
+
+const BUTTON_COLLAPSE_WIDTHS = {
+  first: 1095,
+  second: 1190,
+  third: 1286
+}
+
+// Toast timeouts in ms
+const REPOST_TIMEOUT = 1000
+const SAVED_TIMEOUT = 1000
+
+const messages = {
+  makePublic: 'MAKE PUBLIC',
+  isPublishing: 'PUBLISHING',
+  repostButtonText: 'repost',
+  repostedButtonText: 'reposted',
+  unplayed: 'Unplayed',
+  timeLeft: 'left',
+  played: 'Played',
+  generatedWithAi: 'Generated With AI',
+  actionGroupLabel: 'collection actions'
+}
+
+export type GiantTrackTileProps = {
+  aiAttributionUserId: Nullable<number>
+  artistHandle: string
+  artistName: string
+  badge: Nullable<string>
+  coSign: Nullable<Remix>
+  coverArtSizes: Nullable<CoverArtSizes>
+  credits: string
+  currentUserId: Nullable<ID>
+  description: string
+  doesUserHaveAccess: boolean
+  duration: number
+  fieldVisibility: FieldVisibility
+  following: boolean
+  genre: string
+  isArtistPick: boolean
+  isOwner: boolean
+  isPremium: boolean
+  isPublishing: boolean
+  isRemix: boolean
+  isReposted: boolean
+  isSaved: boolean
+  isUnlisted: boolean
+  listenCount: number
+  loading: boolean
+  mood: string
+  onClickArtistName: () => void
+  onClickFavorites: () => void
+  onClickReposts: () => void
+  onDownload: (trackId: ID, category?: string, parentTrackId?: ID) => void
+  onExternalLinkClick: (event: React.MouseEvent<HTMLAnchorElement>) => void
+  onInternalLinkClick: (url: string) => void
+  onMakePublic: (trackId: ID) => void
+  onFollow: () => void
+  onPlay: () => void
+  onRepost: () => void
+  onSave: () => void
+  onShare: () => void
+  onUnfollow: () => void
+  playing: boolean
+  premiumConditions: Nullable<PremiumConditions>
+  released: string
+  repostCount: number
+  saveCount: number
+  tags: string
+  trackId: number
+  trackTitle: string
+  userId: number
+}
+
+export const GiantTrackTile = ({
+  aiAttributionUserId,
+  artistHandle,
+  artistName,
+  badge,
+  coSign,
+  coverArtSizes,
+  credits,
+  description,
+  doesUserHaveAccess,
+  duration,
+  fieldVisibility,
+  following,
+  genre,
+  isArtistPick,
+  isOwner,
+  isPremium,
+  isRemix,
+  isReposted,
+  isPublishing,
+  isSaved,
+  isUnlisted,
+  listenCount,
+  loading,
+  mood,
+  onClickArtistName,
+  onClickFavorites,
+  onClickReposts,
+  onDownload,
+  onExternalLinkClick,
+  onFollow,
+  onInternalLinkClick,
+  onMakePublic,
+  onPlay,
+  onSave,
+  onShare,
+  onRepost,
+  onUnfollow,
+  released,
+  repostCount,
+  saveCount,
+  playing,
+  premiumConditions,
+  tags,
+  trackId,
+  trackTitle,
+  userId
+}: GiantTrackTileProps) => {
```

Comment:
This was originally the `render` function, and didn't destructure as many props because the helper functions defined on the class would read them directly from `this.props`. Now since we are defining the helpers inline here, they are read from closure and we need to pull more items off of props.